### ISSUE TITLE
NPE after some map source changes (attribution related?)

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#9327]
While playing around with 2020.11.01-NB I changed between Google, OSM and offline map several times.
After going back from Google to OSM offline map the app finally crashed:

```
--------- beginning of crash
11-01 11:52:25.559  5984  5984 E AndroidRuntime: FATAL EXCEPTION: main
11-01 11:52:25.559  5984  5984 E AndroidRuntime: Process: cgeo.geocaching, PID: 5984
11-01 11:52:25.559  5984  5984 E AndroidRuntime: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Class java.lang.Object.getClass()' on a null object reference
11-01 11:52:25.559  5984  5984 E AndroidRuntime: 	at cgeo.geocaching.maps.mapsforge.MapsforgeMapProvider$OfflineMapSource.setMapAttributionTo(MapsforgeMapProvider.java:196)
11-01 11:52:25.559  5984  5984 E AndroidRuntime: 	at cgeo.geocaching.maps.CGeoMap.changeMapSource(CGeoMap.java:1115)
11-01 11:52:25.559  5984  5984 E AndroidRuntime: 	at cgeo.geocaching.maps.CGeoMap.onOptionsItemSelected(CGeoMap.java:899)
11-01 11:52:25.559  5984  5984 E AndroidRuntime: 	at cgeo.geocaching.maps.google.v2.GoogleMapActivity.onOptionsItemSelected(GoogleMapActivity.java:86)
11-01 11:52:25.559  5984  5984 E AndroidRuntime: 	at android.app.Activity.onMenuItemSelected(Activity.java:4182)
11-01 11:52:25.559  5984  5984 E AndroidRuntime: 	at com.android.internal.policy.PhoneWindow.onMenuItemSelected(PhoneWindow.java:1358)
11-01 11:52:25.559  5984  5984 E AndroidRuntime: 	at com.android.internal.view.menu.MenuBuilder.dispatchMenuItemSelected(MenuBuilder.java:787)
11-01 11:52:25.559  5984  5984 E AndroidRuntime: 	at com.android.internal.view.menu.SubMenuBuilder.dispatchMenuItemSelected(SubMenuBuilder.java:85)
11-01 11:52:25.559  5984  5984 E AndroidRuntime: 	at com.android.internal.view.menu.MenuItemImpl.invoke(MenuItemImpl.java:157)
11-01 11:52:25.559  5984  5984 E AndroidRuntime: 	at com.android.internal.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:934)
11-01 11:52:25.559  5984  5984 E AndroidRuntime: 	at com.android.internal.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:924)
11-01 11:52:25.559  5984  5984 E AndroidRuntime: 	at com.android.internal.view.menu.MenuPopup.onItemClick(MenuPopup.java:128)
11-01 11:52:25.559  5984  5984 E AndroidRuntime: 	at android.widget.AdapterView.performItemClick(AdapterView.java:374)
11-01 11:52:25.559  5984  5984 E AndroidRuntime: 	at android.widget.AbsListView.performItemClick(AbsListView.java:1736)
11-01 11:52:25.559  5984  5984 E AndroidRuntime: 	at android.widget.AbsListView$PerformClick.run(AbsListView.java:4207)
11-01 11:52:25.559  5984  5984 E AndroidRuntime: 	at android.widget.AbsListView$7.run(AbsListView.java:6692)
11-01 11:52:25.559  5984  5984 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:883)
11-01 11:52:25.559  5984  5984 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:100)
11-01 11:52:25.559  5984  5984 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:237)
11-01 11:52:25.559  5984  5984 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8154)
11-01 11:52:25.559  5984  5984 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
11-01 11:52:25.559  5984  5984 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
11-01 11:52:25.559  5984  5984 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)
```

**Comment by eddiemuc** [comment]
This is clearly related to my attribution changes. However, code base is changed significantly currently due to new GUi approach ("Locus"-like), so I guess it does not make much sense to fix this now.

**Comment by eddiemuc** [comment]
Strange though how this could happen. See here the relevant lines:

https://github.com/cgeo/cgeo/blob/8d1eb842f2858125ea300c3d80661b1e725ce5c6/main/src/cgeo/geocaching/maps/mapsforge/MapsforgeMapProvider.java#L192-L209

**Comment by Lineflyer** [comment]
Found a way to reproduce:
- Start c:geo
- Change to Google Satellite
- End c:geo (forced to have clean conditions)
- Start c:geo
- Scroll away from own position
- Change map to OSM offline map (In my case germany openandromaps)
- Crash

**Comment by eddiemuc** [comment]
Will be fixed with PR #9323
